src/detect-flowint.c

```diff
@@ -230,27 +230,28 @@ static DetectFlowintData *DetectFlowintParse(DetectEngineCtx *de_ctx, const char
     char *varname = NULL;
     char *varval = NULL;
     char *modstr = NULL;
-    int ret = 0, res = 0;
+    int res = 0;
     size_t pcre2_len;
     uint8_t modifier = FLOWINT_MODIFIER_UNKNOWN;
     unsigned long long value_long = 0;
     const char *str_ptr;
+    pcre2_match_data *match = NULL;
 
-    ret = DetectParsePcreExec(&parse_regex, rawstr, 0, 0);
+    int ret = DetectParsePcreExec(&parse_regex, &match, rawstr, 0, 0);
     if (ret < 3 || ret > 4) {
         SCLogError("\"%s\" is not a valid setting for flowint(ret = %d).", rawstr, ret);
         return NULL;
```

Comment:
leaks match if ret >= 0 but not 3 or 4, fixing

Reply:
Good catch; fixing now.